File: 2. Hash Maps and Sets/128. Longest Consecutive Sequence/Solution.py
from typing import List

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums: return 0

        longest_chain = 0
        num_set = set(nums)

        for num in num_set:

            if num - 1 not in num_set:
                curr_num = num
                curr_chain = 1

                while curr_num + 1 in num_set:
                    curr_num += 1
                    curr_chain += 1
                
                longest_chain = curr_chain if curr_chain > longest_chain else longest_chain
        
        return longest_chain

File: 2. Hash Maps and Sets/128. Longest Consecutive Sequence/Solution.cpp
#include <vector>
#include <unordered_set>

class Solution {
public:
    int longestConsecutive(std::vector<int>& nums) {
        if (!nums.size()) return 0;
        int longest_chain = 0;

        std::unordered_set<int> num_set;
        for (auto& num : nums)
        {
            num_set.insert(num);
        }

        for (auto& num : num_set)
        {
            if (!num_set.count(num - 1))
            {
                int curr_num = num;
                int curr_chain = 1;

                while (num_set.count(curr_num + 1))
                {
                    curr_num += 1;
                    curr_chain += 1;
                }

                if (longest_chain < curr_chain) longest_chain = curr_chain;
            }
        }

        return longest_chain;
    }
};